lua/treesitter-context.lua

```diff
@@ -368,6 +368,10 @@ local function get_parent_matches(max_lines)
     return
   end
 
+  if config.separator then
+    max_lines = max_lines - 1 -- separator takes 1 line
```

Comment:
This does not account for when `max_lines == 1`.

Reply:
@lewis6991 When separator is set and there is only 1 line above cursorline, there would be no space for displaying both context line and separator. I think it's fine to show nothing in this situation.